cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/AutoscaleRecommendation.java

```diff
@@ -0,0 +1,41 @@
+package com.sequenceiq.cloudbreak.cloud.model;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class AutoscaleRecommendation {
+
+    private final Set<String> hostGroups;
```

Comment:
Splitting this into loadBasedScaling \ timeBasedScaling will be good.

Reply:
Also will AutoscaleRecommendation be exposed as top level api BlueprintUtilV4Endpoint::getAutoscaleRecommendation(workspaceId,blueprintName) for validation in periscope against Autoscale CLI requests?  Something similar to BlueprintUtilV4Endpoint::getServicesByBlueprint. BlueprintUtilV4Endpoint::createRecommendation will not be usable from periscope since parameters like credentialName, region are not available.

Reply:
Split is done.

Reply:
[This](https://gk1-console.cdp-priv.mow-dev.cloudera.com/cloud/cloudbreak/v4/0/blueprints_util/recommendation?blueprintName=7.1.0%20-%20Data%20Engineering:%20HA:%20Apache%20Spark,%20Apache%20Hive,%20Apache%20Oozie&credentialName=gk1cred&region=us-west-2&resourceType=DATAHUB) returns,

```
"autoscaleRecommendation":{"timeBasedHostGroups":["compute","gateway"],"loadBasedHostGroups":["compute"]},"resizeRecommendation":{"scaleUpHostGroups":["worker","gateway","compute"],"scaleDownHostGroups":["worker","gateway","compute"]}}
```

I will check the generated swagger definition and see what apis to add.

Reply:
Yes this response looks good. All autoscaling related CB API  are accessed  via AutoscaleV4Endpoint. It would be nice if this is also exposed in that.

Reply:
I will add the endpoint via a new PR.